### PR TITLE
Cleaned Up PR: Removed Unwanted Commits and Refined Changes for Prediction Guard Comps

### PR DESCRIPTION
**Pull request by sharanshirodkar7**
## Description

This PR provides a cleaned-up version of the previous PR by removing unwanted commits that were mistakenly included. The relevant changes and improvements made by Prediction Guard have been preserved. Key changes in this PR include:

	•	Adding Prediction Guard microservices and reorganizing guardrails.
	•	Removing legacy LLava folders and updating to PG embedding models.
	•	Addressing security issues and refactoring the hashing function.
	•	Adding unit tests for various microservices (e.g., embeddings, LLM, LVM).
	•	Updating pre-commit configurations and resolving docker file reference warnings.

The motivation behind this cleanup is to provide a clearer commit history and ensure only relevant changes are reviewed and merged.

## Issues

n/a

## Type of change

- [ ] Others (enhancement, documentation, validation, etc.)

## Dependencies

No new 3rd party dependencies introduced.

## Tests

	•	All relevant unit tests have been added for new microservices.
	•	Pre-commit checks have been updated and validated to ensure code quality.
	•	Manual testing performed to verify that Docker configurations and microservice integrations work as expected.

### CODE REVIEW

File: comps/embeddings/predictionguard/test_service.py
Comment (lvliang-intel) [inline comps/embeddings/predictionguard/test_service.py:1]: Please move the test code to 'https://github.com/opea-project/GenAIComps/tree/main/tests'.

Comment (sharanshirodkar7) [reply comps/embeddings/predictionguard/test_service.py:1]: Will do.

Comment (sharanshirodkar7) [reply comps/embeddings/predictionguard/test_service.py:1]: Have moved all the test codes to the folder you linked @lvliang-intel

**Comment by sharanshirodkar7** [comment]
@lvliang-intel 

Note: This PR has a dependency on the examples repository PR: https://github.com/opea-project/GenAIExamples/pull/732
During the process, we encountered several formatting issues flagged by the prettier hook for various files that were not modified by us. We are unsure if it’s our responsibility to address these formatting issues. Could you please provide guidance on whether we need to fix these bugs, or if it would be better handled in a separate PR by the team responsible for those files?

Looking forward to your suggestions.

**Comment by sharanshirodkar7** [comment]
@lvliang-intel For testing our microservices, the `PREDICTIONGUARD_API_KEY` is required. How should we proceed with this? Who should we provide the API key to so that it can be securely stored in Git secrets or a similar solution?

**Comment by chensuyue** [comment]
> @lvliang-intel For testing our microservices, the `PREDICTIONGUARD_API_KEY` is required. How should we proceed with this? Who should we provide the API key to so that it can be securely stored in Git secrets or a similar solution?

Please send me by teams.

**Comment by sharanshirodkar7** [comment]
@chensuyue please let me know here when you accept my invite to message on teams. My mail id : `sharan@predictionguard.com`

**Comment by chensuyue** [comment]
> @chensuyue please let me know here when you accept my invite to message on teams. My mail id : `sharan@predictionguard.com`

Sorry, I didn't find any email or teams message. Would you like to send me the token directly by email?

**Comment by sharanshirodkar7** [comment]
@chensuyue I have just sent you an email with the api key.

**Comment by chensuyue** [comment]
> @chensuyue I have just sent you an email with the api key. Also PR is ready for review : `https://github.com/opea-project/GenAIExamples/pull/732`

https://github.com/opea-project/GenAIComps/pull/650

File: .github/workflows/docker/compose/llms-compose.yaml [hunk 1]
```diff
@@ -36,3 +36,7 @@ services:
     build:
       dockerfile: comps/llms/text-generation/vllm-ray/docker/Dockerfile.vllmray
     image: ${REGISTRY:-opea}/llm-vllm-ray-hpu:${TAG:-latest}
+  llm-predictionguard:
```
Comment (chensuyue) [inline .github/workflows/docker/compose/llms-compose.yaml:39]: no need to add image build here.

File: .github/workflows/docker/compose/guardrails-compose.yaml [hunk 2]
```diff
@@ -9,3 +9,28 @@ services:
     build:
       dockerfile: comps/guardrails/llama_guard/docker/Dockerfile
     image: ${REGISTRY:-opea}/guardrails-tgi:${TAG:-latest}
```
Comment (chensuyue) [inline .github/workflows/docker/compose/guardrails-compose.yaml:11]: `guardrails-compose.yaml` include the images used by `GenAIExamples`, `guardrails-compose-cd.yaml` include the other images not used by `GenAIExamples`. No need to duplicate images in those 2 files.

**Comment by chensuyue** [comment]
Need to adapt to the new code structure.

**Comment by sharanshirodkar7** [comment]
@chensuyue yes I feel its better that we have separate PR for each comps. I will do that.

**Pull request closed without merging**